### PR TITLE
feature: folder dto validation

### LINKED ISSUE #4684: add DTO validation for folder name, on folder creation
### Is your proposal related to a problem?

The folder name (on folder creation) validation should have a data transfer object.

### Describe the solution you'd like

Create a DTO ([example](https://github.com/ToolJet/ToolJet/blob/develop/server/src/dto/version-create.dto.ts)) for validation for folder name (on folder creation)

reference: https://github.com/ToolJet/ToolJet/pull/4402#discussion_r1005465650

### PR DESCRIPTION
**Pull request by mikebarr24**
Resolves #4684
I think this is what was needed. Happy to take feedback and any edits. 😀

### CODE REVIEW

File: server/src/controllers/folders.controller.ts [hunk 1]
```diff
@@ -41,7 +41,9 @@ export class FoldersController {
     if (!ability.can('updateFolder', Folder)) {
       throw new ForbiddenException('You do not have permissions to perform this action');
     }
-    const folder = await this.foldersService.update(id, folderName);
+    const createFolderDto = new CreateFolderDto();
```
Comment (gsmithun4) [inline server/src/controllers/folders.controller.ts:44]: Should do the changes here
https://github.com/ToolJet/ToolJet/pull/4849/files#diff-40c2ee9b56c32bbadbd17595c66c93b6689b59e2431284a343b14f3ea69abe81L38
body should be accepted as a `createFolderDto` object

File: server/src/controllers/folders.controller.ts [hunk 2]
```diff
@@ -27,9 +27,9 @@ export class FoldersController {
     if (!ability.can('createFolder', Folder)) {
       throw new ForbiddenException('You do not have permissions to perform this action');
     }
-    const folderName = req.body.name;
+    createFolderDto.name = req.body.name;
```
Comment (gsmithun4) [inline server/src/controllers/folders.controller.ts:30]: Why should we need to do this? `name` will be already available in `createFolderDto`

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: server/src/controllers/folders.controller.ts (hunk 1 above)
Comment (mikebarr24) [reply server/src/controllers/folders.controller.ts]: I think I get you for both these comments! Thank you! Still trying to get my head round DTOs

File: server/src/controllers/folders.controller.ts (hunk 2 above)
Comment (mikebarr24) [reply server/src/controllers/folders.controller.ts]: I think I get you for both these comments! Thank you! Still trying to get my head round DTOs

**Comment by mikebarr24** [comment]
I think I get you for both these comments! Thank you! Still trying to get my head round DTOs

**Comment by mikebarr24** [comment]
Thank you so much! Hope this is OK.

**Review by gsmithun4 — DISMISSED** [review]
LGTM

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**